src/vss_tools/vspec/vssexporters/vss2samm/helpers/vssHelper.py

```diff
@@ -0,0 +1,569 @@
+# REASON FOR THIS is that:
+#       VSSNode.children is immutable, when the node had some children
+#       BUT: we can set (add) them, in case when VSSNode.children was initially empty.
+def clone_vss_node(vss_node: VSSNode):
+    # NOTE: Below properties names are taken from: VSSNode: validate_vss_element function.
+    #       If required to sync these, use the aforementioned function as reference.
+    #
+    #       Currently we skip the "children" one,
+    #       as these will be further filtered and populated in the cloned_vss_node.
+    vss_node_dict = {}
+
+    # key:value pairs where:
+    #     key   - is the VSSNode property
+    #     value - is the VSS specification, property,
+    #             which was used for the creation of vss_node,
+    #             i.e. its corresponding name in the used for creation of VSSNode, dictionary.
+    properties_to_clone = {
+        "$file_name$"   : "$file_name$",    # noqa: E203
+        "type"          : "type",           # noqa: E203
+        "datatype"      : "datatype",       # noqa: E203
+        "description"   : "description",    # noqa: E203
+        "unit"          : "unit",           # noqa: E203
+        "uuid"          : "uuid",           # noqa: E203
+        "min"           : 'min',            # noqa: E203
+        "max"           : "max",            # noqa: E203
+        "aggregate"     : "aggregate",      # noqa: E203
+        "default"       : "default",        # noqa: E203
+        "instances"     : "instances",      # noqa: E203
+        "deprecation"   : "deprecation",    # noqa: E203
+        "arraysize"     : "arraysize",      # noqa: E203
+        "comment"       : "comment",        # noqa: E203
+        "allowed"       : "allowed"         # noqa: E203
+    }
+
+    for vss_prop_name in properties_to_clone.keys():
+        if vss_prop_name == "default" and vss_node.type not in [VSSType.ATTRIBUTE, VSSType.ACTUATOR, VSSType.SENSOR]:
+            # Skip processing for "default" fields for NON ATTRIBUTE, ACTUATOR and SENSOR nodes
+            continue
+
+        if hasattr(vss_node, vss_prop_name):
+            vss_prop_value = vss_node.__getattribute__(vss_prop_name)
+            vss_dict_prop_name = properties_to_clone[vss_prop_name]
+
+            if type(vss_prop_value) in [VSSType, VSSDataType]:
+                vss_node_dict[vss_dict_prop_name] = vss_prop_value.value
+            else:
+                vss_node_dict[vss_dict_prop_name] = vss_prop_value
+
+    return VSSNode(vss_node.name, vss_node_dict, VSSTreeType.SIGNAL_TREE.available_types())
+
+
+# Helper function, to check whether to use parent prefix for a vss_node or not
+def should_use_parent_prefix(vss_node: VSSNode):
+    if (top_vss_tree_unique_node_names[vss_node.name]['counter'] > 1) \
+        or (vss_node.is_leaf
+            and hasattr(vss_node, 'datatype')
+            and vss_node.datatype is VSSDataType.BOOLEAN
+            and (vss_node.name.startswith('Is') or vss_node.name.startswith('is'))
```

Comment:
`vss_node.name.lower().startswith("is")`

Reply:
Done